File: insfrastructure/usecases/get_all_accounts_usecase.py
from insfrastructure.entities.bank_account_entity import BankAccountEntity
from insfrastructure.repositories.bank_account_repository import BankAccountRepository


class GetAllAccountsUseCase:
    def __init__(self, bank_account_repo: BankAccountRepository):
        self.bank_account_repo = bank_account_repo

    def execute(self) -> list[BankAccountEntity]:
        return self.account_repository.show_all_accounts()

File: typescript-impl/src/infrastructure/usecases/get_all_accounts_usecase.ts
import { BankAccountRepository } from '../repositories/bank_account_repository';
import { BankAccountEntity } from '../entities/bank_account_entity';

export class GetAllAccountsUseCase {
    constructor(private bank_account_repo: BankAccountRepository) { }

    execute(): BankAccountEntity[] {
        return this.bank_account_repo.show_all_accounts();
    }
}
